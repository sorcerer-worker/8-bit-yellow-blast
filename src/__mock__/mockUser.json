{
    "characterName": "Test User",
    "discordId": "757938182483485954",
    "class": "warrior",
    "pouch": 98638,
    "pvp": false,
    "inventory": {
      "weapons": [],
      "consumables": [],
      "classSkills": [],
      "otherSkills": []
    },
    "health": {
      "currentHealth": 10,
      "maxHealth": 10
    },
    "level": {
      "currentLevel": 1
    },
    "stats": {
      "strength": 5,
      "defense": 5,
      "speed": 28,
      "vitality": 10,
      "intelligence": 5,
      "magicResistance": 5
    }
  }