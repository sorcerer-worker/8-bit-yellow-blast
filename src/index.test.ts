import { Battle } from "./index";

// json files
import mockUser from "./__mock__/mockUser.json"
import mockConfig from "./__mock__/mockConfig.json"

let bat: Battle;

describe('battle system', () => {
    beforeAll(() => {
        bat = new Battle(mockUser, mockUser, mockConfig);
    })
    it('makes a battle', () => {
        expect(bat).toBeInstanceOf(Battle);
    })
    it('has a user attack', () => {
        expect(bat.attack(bat.user1.name, "Punch")).toBe("Test User attacked with Punch");
    })
    it('has turn system integrated into the battle system', () => {
        expect(bat.turns.users).toMatchSnapshot();
    })
})