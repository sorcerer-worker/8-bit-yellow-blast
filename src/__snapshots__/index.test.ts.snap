// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`battle system can use getUserById 1`] = `
User {
  "data": Object {
    "class": "warrior",
    "health": Object {
      "currentHealth": 10,
      "maxHealth": 10,
    },
    "id": 757938182483486000,
    "inventory": Object {
      "classSkills": Array [],
      "consumables": Array [],
      "otherSkills": Array [],
      "weapons": Array [],
    },
    "isTurn": true,
    "level": Object {
      "currentLevel": 1,
    },
    "name": "Test User",
    "pouch": 98638,
    "pvp": false,
    "stats": Object {
      "defense": 5,
      "intelligence": 5,
      "magicResistance": 5,
      "speed": 28,
      "strength": 5,
      "vitality": 10,
    },
  },
}
`;

exports[`battle system has integrated turn system 1`] = `
Object {
  "currentTurn": 0,
  "turns": Map {
    0 => "Test User",
    1 => "Test User 2",
  },
}
`;

exports[`battle system has turn system integrated into the battle system 1`] = `
Object {
  "currentTurn": 0,
  "turns": Map {
    0 => "Test User",
    1 => "Test User 2",
  },
}
`;
